src/hooks/use-supervisors.ts

```diff
@@ -0,0 +1,37 @@
+import { useAtom } from "jotai";
+import { useCallback } from "react";
+
+import { supervisorsAtom } from "@/atoms/supervisors";
+import type { SavedSupervisor } from "@/types/supervisor";
+
+export function useSupervisors() {
+  const [supervisors, setSupervsiors] = useAtom(supervisorsAtom);
```

Comment:
**issue:** typo, needs to be changed here and in `addSupervisor` and `removeSupervisor` callbacks